src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/DisposableBox.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests
+{
+    internal class DisposableBox<T> : IDisposable where T : class, IDisposable
+    {
+        public T Value { get; private set; }
+
+        public DisposableBox(T value)
+        {
+            Value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        public void Dispose()
+        {
+            Release()?.Dispose();
+        }
+
+        public async Task<U> GetAndReleaseAsync<U>(Func<T, Task<U>> func)
+        {
+            U value = await func(Value).ConfigureAwait(false);
```

Comment:
Can you add some comments to this class? It would help to understand how it transfers lifetime ownership.